compiler/Core/OptimizerException.h

```diff
@@ -0,0 +1,14 @@
+#include "common/Exception.h"
+
+namespace sorbet::compiler {
+// Occasionally we need to throw errors from within LLVM optimization passes,
+// where we don't have access to GlobalState and therefore cannot use failCompilation.
+// We have this separate class so that the compiler plugin can differentiate this
+// exception from AbortCompilation and know that a Sorbet-level error needs to be
+// emitted.  The user can therefore have some idea of why Sorbet failed.
+class OptimizerException : public sorbet::SorbetException {
```

Comment:
Should this use inheritance? Or should it define its own hierarchy?

Reply:
ah looks like we use inheritance when we define AbortCompilation. Makes sense to keep it the same then.